tensorflow-logging/src/TensorFlow/Logging.hs

```diff
@@ -151,16 +154,27 @@ doubleWallTime = asDouble <$> getCurrentTime
 -- | Adds a 'CoreOps.histogramSummary' node. The tag argument is intentionally
 -- limited to a single value for simplicity.
 histogramSummary ::
-    (MonadBuild m, TensorType t, t /= ByteString, t /= Bool)
-     -- OneOf '[Int16, Int32, Int64, Int8, Word16, Word8, Double, Float] t)
+    (MonadBuild m, OneOf '[Int16, Int32, Int64, Int8, Word16, Word8, Double, Float] t)
     => ByteString -> Tensor v t -> m ()
 histogramSummary tag = addSummary . CoreOps.histogramSummary (scalar tag)
 
+-- | Adds a 'CoreOps.imageSummary' node. The tag used in the summary is
+-- | computed from label <> step.
+imageSummary ::
+    (OneOf '[Word8, Word16, Float] t, MonadBuild m)
+    => ByteString -- label
+    -> Tensor v1 Int64 -- step
```

Comment:
Does it make sense to include step in the tag automatically? That seems like something the caller can do themselves.